src/types-and-traits.rst

```diff
@@ -1995,6 +1995,39 @@ obtained as follows:
    #. :dp:`fls_ju4ypa5ysga0`
       Continue with the next source :t:`type`.
 
+
+.. _fls_emcNJzl2tHSA:
+
+Structural Equality
+~~~~~~~~~~~~~~~~~~~
+
+.. rubric:: Legality Rules
+
+:dp:`fls_uVTpA7gbLCYX`
+A :t:`structurally equal` :t:`type` can be :t:`pattern matched <pattern
+matching>` on via the use of an :t:`associated constant` or :t:`constant`.
+
+:dp:`fls_emcNJzl2tHSA`
+:t:`[Bool]`, :t:`Char`, :t:`[integer type]s`, :t:`str`, :t:`[function pointer
```

Comment:
The Bool, char, and str references should probably use :c: to get the code-like formatting.